dsgrid/registry/registry_database.py

```diff
@@ -1,415 +1,643 @@
-                RETURN e.version
-        """,
-            count=True,
+        table = self.get_table(RegistryTables.REGISTRATIONS)
+        res = conn.execute(
+            insert(table).values(
+                timestamp=str(registration.timestamp),
+                submitter=registration.submitter,
+                update_type=registration.update_type,
+                log_message=registration.log_message,
+            )
         )
-        count = cursor.count()
-        assert count <= 1, f"{db_id=} {count=}"
-        if count == 0:
-            raise DSGValueNotRegistered(f"{db_id=} is not registered")
-        return cursor.next()
-
-    def has(self, db_id, version=None):
-        collection_name, key = db_id.split("/")
-        if version is None:
-            return self._db.collection(collection_name).has(key)
-
-        cursor = self._db.aql.execute(
-            f"""
-            FOR v, e, p in 0..{MAX_CONFIG_VERSIONS}
-                OUTBOUND "{db_id}"
-                GRAPH "{GRAPH}"
-                OPTIONS {{edgeCollections: ["{Edge.UPDATED_TO.value}"]}}
-                FILTER e.version == "{version}"
-                RETURN p.vertices[1]
-        """,
-            count=True,
+        data = registration.model_dump(mode="json")
+        data["id"] = res.lastrowid
+        return RegistrationModel(**data)
+
+    def get_registration(self, conn: Connection, db_id: int) -> RegistrationModel:
+        """Return the registration information for the database ID."""
+        table1 = self.get_table(RegistryTables.MODELS)
+        table2 = self.get_table(RegistryTables.REGISTRATIONS)
+        stmt = (
+            select(
+                table2.c.id,
+                table2.c.timestamp,
+                table2.c.submitter,
+                table2.c.update_type,
+                table2.c.log_message,
+            )
+            .join_from(table1, table2, table1.c.registration_id == table2.c.id)
+            .where(table1.c.id == db_id)
         )
-        return cursor.count() > 0
-
-    def insert_contains_edge(self, db_id1, db_id2):
-        self._graph.edge_collection(Edge.CONTAINS.value).insert(
-            {
-                "_from": db_id1,
-                "_to": db_id2,
-            }
+        res = conn.execute(stmt).fetchall()
+        if not res:
+            msg = f"{db_id=}"
+            raise DSGValueNotRegistered(msg)
+        if len(res) > 1:
+            msg = f"Bug: found more than one row matching {db_id=}"
+            raise Exception(msg)
+        row = res[0]
+        return RegistrationModel(
+            id=row[0],
+            timestamp=row[1],
+            submitter=row[2],
+            update_type=row[3],
+            log_message=row[4],
         )
 
-    def insert_latest_edge(self, root, model):
-        self._graph.edge_collection(Edge.LATEST.value).insert(
-            {
-                "_from": root.id,
-                "_to": model.id,
-            }
+    def get_initial_registration(
```

Comment:
Is there a "get all registrations"? This method struck me a little funny, but I can see how the first registration might be extra-important. There could be a get_nth_registration ...

Reply:
This actually isn't used in the code - except for a test I added for coverage. I added it as I was porting the Arango code. (We stored this information but didn't use it.) I almost deleted it, but it seemed like it might be useful. I thought about adding a `list_registrations` that sorted the registrations by date. I decided to wait to see if people actually need it. (And wanted to be done.) If the users are us, we may prefer viewing this information in sqlite.